### ISSUE TITLE
Missing POSIX confstr and _CS_* symbols on Linux.

### ISSUE THREAD

**Issue by jgarzik** [rust-lang/libc#3767]
Apple Darwin provides the POSIX standard `confstr(3)` function and associated `_CS_*` symbols.  Tested on Darwin and is working.

Requesting the same for Linux.

Quoting src/unix/bsd/apple/mod.rs:
```
// `confstr` keys (only the values guaranteed by `man confstr`).
pub const _CS_PATH: ::c_int = 1;
pub const _CS_DARWIN_USER_DIR: ::c_int = 65536;
pub const _CS_DARWIN_USER_TEMP_DIR: ::c_int = 65537;
pub const _CS_DARWIN_USER_CACHE_DIR: ::c_int = 65538;
...
    #[cfg_attr(
        all(target_os = "macos", target_arch = "x86"),
        link_name = "confstr$UNIX2003"
    )]
    pub fn confstr(name: ::c_int, buf: *mut ::c_char, len: ::size_t) -> ::size_t
```

The Linux version for this crate should be very similar.  The Linux man page for `confstr(3)` also helpfully lists the few key `_CS_*` symbols that the developer should expect to be available:
```
       _CS_GNU_LIBC_VERSION (GNU C library only; since glibc 2.3.2)
       _CS_GNU_LIBPTHREAD_VERSION (GNU C library only; since glibc 2.3.2)
       _CS_PATH
```

**Comment by devnexen** [comment]
Would you be pleased to create a PR ?

**Comment by jgarzik** [comment]
@devnexen I gave it a shot with #3771, but I do not understand why the arm test is failing, so I'm a bit stuck.

**Comment by magicant** [comment]
For the libc-0.2 branch, #3612 is awaiting review.

**Comment by jgarzik** [comment]
> For the libc-0.2 branch, #3612 is awaiting review.

Would it make sense to push to the main branch first, then backport once merged?

(I'm not familiar with this project's upstreaming flows)

**Comment by magicant** [comment]
I opened #3612 in the hope that 0.2.x would get released sooner than 1.0. If 1.0 is coming out soon, I'd retarget for the main branch. Any plans for future releases?

**Comment by jgarzik** [comment]
> I opened #3612 in the hope that 0.2.x would get released sooner than 1.0. If 1.0 is coming out soon, I'd retarget for the main branch. Any plans for future releases?

Maybe I'm an old greybeard, but usually the norm is to post an upstream patch first, then backport to a branch once merged upstream, regardless of date when the next upstream release will come?
